[WIP] Capstone v6 support

jiegec opened the pull request:
Tracking upstream capstone v6 development. Do not merge.

jiegec committed:
sys: bump capstone to 6.0.0-alpha4, fix arm64 and sysz rename

jiegec committed:
rs: rename arm64 -> aarch64 and sysz -> systemz to follow upstream

jiegec committed:
rs: adapt to capstone v6 changes, mainly in aarch64

jiegec committed:
rs: use aarch64_reg::Type for windows compat

jiegec committed:
rs: run cargo fmt to make ci happy

jiegec committed:
rs, sys: mention capstone 6.0.0-alpha4 and arch renaming in CHANGELOG

jiegec committed:
rs: avoid possible panics in library functions

jiegec committed:
rs: add missing doc comments

jiegec committed:
rs, sys: generate arm_spsr_cspr_bits as bitfield enum

jiegec committed:
rs: add previously failed testcase from issue 175

jiegec committed:
ci: rename arch_arm64 to arch_aarch64 feature

jiegec committed:
sys: add Alpha arch support

jiegec committed:
rs, sys: fix feature detection after renaming

jiegec committed:
sys: add HPPA arch support

jiegec committed:
sys: add LoongArch arch support

jiegec committed:
sys: add Xtensa arch support

jiegec committed:
sys: add ARC arch support

jiegec committed:
sys: move bpf bindings around to become in order

jiegec committed:
rs: add Alpha arch support

jiegec committed:
sys: fix build.rs lifetime for Rust 1.70.0

jiegec committed:
rs: add HPPA arch support

jiegec committed:
rs: add LoongArch arch support

jiegec committed:
rs: rename variable to reflect its type

jiegec committed:
rs: disable failing hppa test due to upstream bug

jiegec committed:
rs: add Xtensa arch support

jiegec committed:
rs: fix tests on windows

jiegec committed:
rs: add ARC arch support

jiegec committed:
rs: fix type conversion on windows

jiegec committed:
rs: bump capstone to 6.0.0-Alpha4-8-gefc0ba44 to fix upstream bugs

jiegec committed:
rs: update tests after upstream bugs are fixed

jiegec committed:
rs: test CS_OPT_UNSIGNED using x86 code

jiegec committed:
rs, sys: bump capstone to 6.0.0-Alpha4-13-gfe6bdc6e

jiegec committed:
sys: bump capstone to 6.0.0-Alpha4-25-g717d8b05

jiegec committed:
rs: bump capstone to 6.0.0-Alpha4-25-g717d8b05

jiegec committed:
rs: add access mode to sparc operand

jiegec committed:
rs: document recent changes in CHANGELOG

jiegec committed:
rs: update tests for sparc, not a bug actually

jiegec committed:
rs: apply cargo clippy

jiegec committed:
rs: fix tests on windows

jiegec committed:
rs, sys: bump capstone to 6.0.0-Alpha5

jiegec committed:
rs: zero initialize cs_insn to make valgrind happy

jiegec committed:
sys: regenerate bindings to use ffi instead of libc for c_uint type

jiegec committed:
rs: use ffi instead of libc for c_uint type

jiegec committed:
rs: drop misplaced changelog due to git rebase

jiegec committed:
rs, sys: bump capstone to 6.0.0-Alpha6

jiegec committed:
rs, sys: bump capstone to 6.0.0-Alpha7

Add support for RISCV now operand types: FP and CSR.

tmfink commented:
It looks like cargo failing in curl is mentioned in:
https://github.com/rust-lang/cargo/issues/16357
https://github.com/alexcrichton/curl-rust/issues/632

It seems like we can work around this in are CI infra by upgrading our FreeBSD version to 15.0.

jiegec committed:
rs, sys: bump capstone to 6.0.0-Alpha8

CodeWenjiu committed:
sys: only compile enabled arch sources in build.rs

In GCC 14+, -Wimplicit-function-declaration became an error that -w flag
cannot suppress. The auto-generated AArch64 sources in capstone v6
trigger
this error even when the AArch64 feature is disabled, because build.rs
previously compiled all arch directories unconditionally.

Filter arch directories by enabled cargo features, so only selected
architectures' sources are compiled.

CodeWenjiu committed:
sys: panic on unhandled arch dir, allow clippy match_like_matches_macro

CodeWenjiu committed:
sys: fix panic format for Rust 2018 edition compatibility

CodeWenjiu committed:
sys: remove redundant clippy allow attribute

jiegec committed:
rs, sys: bump capstone to 6.0.0-Alpha10

jiegec committed:
sys: fix cargo clippy warnings

jiegec committed:
rs: do not hardcode csr number, use enum riscv_sysreg instead
